keps/sig-architecture/3935-oldest-node-newest-control-plane/README.md

```diff
@@ -0,0 +1,799 @@
+
+### Goals
+
+* Allow users to make full use of the [yearly support period](https://github.com/kubernetes/enhancements/tree/master/keps/sig-release/1498-kubernetes-yearly-support-period)
+* Streamline upgrade paths for nodes / node pools to minimize required workload churn
+* Validate oldest supported node components work with latest control plane components
+* Update the [Kubernetes version skew policy](https://kubernetes.io/releases/version-skew-policy/) to indicate node components from the oldest supported minor version work with control plane components from the newest supported minor version, starting with Kubernetes `v1.25` nodes and `v1.28` control planes.
+
+### Non-Goals
+
+* This does not propose any changes to the number of supported minor versions
+* This does not propose any changes to the minor version support duration
+* This does not propose any changes to the Kubernetes release process
+* This does not propose any changes to the mechanics control plane components use to enable features 
+  and work with older nodes without features ([KEP-3920](https://github.com/kubernetes/enhancements/pull/3920) could help with this)
+* This does not propose delaying introduction of existing features SIGs have planned around existing n-2 node/control plane skew
+* This does not address increasing node skew supported by cluster management tools like `kubeadm`. Maintainers of those tools
+  can choose to make use of the additional supported skew if they wish, but are not required to. For example, `kubeadm` currently officially
+  supports managing [kubelets up to one version older](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/#version-skew-policy)
+  than the `kubeadm` instance, and does not make use of the existing supported n-2 skew in upgrade plans.
+* This does not address third-party components that version themselves with node components and limit skew relative to the control plane.
+  Maintainers of those components can choose to support larger or smaller version skew. Node operators should limit themselves to skew
+  supported by core Kubernetes node components and any third-party components required for node operation.
+
+## Proposal
+
+1. Add a configuration to https://testgrid.k8s.io/sig-node-kubelet#Summary to begin testing 3 version node / control plane skew
+2. Identify in-flight feature enablement plans from SIGs that rely on a maximum of 2 version node / control plane skew, and the node version when those plans would be complete
+3. Choose the minimum node version to begin supporting 3-version skew that avoids disrupting the plans identified in step 2
+   * 1.25 nodes / 1.28 control planes
+4. Update the [Kubernetes version skew policy](https://kubernetes.io/releases/version-skew-policy/) as part of the release identified in step 3
+
+### Risks and Mitigations
+
+`kubeadm` currently limits itself to managing nodes a single minor version older than the `kubeadm` build.
```

Comment:
not closely related, but somehow related is CRI API  (specifically crictl) versioning. See https://github.com/kubernetes/kubernetes/pull/114394/files. Likely we need to follow the same on the Node so crictl must support minimal supported CRI version.

Reply:
agree @SergeyKanzhelev

Reply:
thanks for the link, I'll add that in a `related work` section as a breadcrumb for people

Reply:
CC: @mrunalp